slick/src/main/scala-2/slick/lifted/MappedTo.scala

```diff
@@ -0,0 +1,54 @@
+package slick.lifted
+
+import scala.language.experimental.macros
+import scala.reflect.macros.blackbox.Context
+import scala.util.control.NonFatal
+
+/** An isomorphism between two types that can be used for mapped column types. */
+class Isomorphism[A, B](val map: A => B, val comap: B => A)
```

Comment:
Perhaps the common parts could be moved to a shared source directory. Although MappedToBase would need to be in the same file as its version-dependent companion.

Reply:
Is Isomorphism supposed to be instantiated by users ever or is it just an implementation detail of the macro?

If the former, perhaps it would be more ergonomic to either change the constructor, or provide an `apply` method, that uses two argument lists to support type inference.